src/contexts/QuizModalContext.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { useContext, createContext, useState } from "react"
+
+export const QuizModalContext = createContext<{
+  isOpen: boolean
+  setIsOpen: (value: boolean) => void
+}>({
+  isOpen: false,
+  setIsOpen: () => {},
+})
+
+export const QuizModalContextProvider: React.FC = ({ children }) => {
+  const [isOpen, setIsOpen] = useState(false)
+
+  console.log("the wquiz contet ", isOpen)
+
+  return (
+    <QuizModalContext.Provider value={{ isOpen, setIsOpen }}>
```

Comment:
Maybe we can add `open` and `close`. I saw we have to use `setIsOpen(false/true)` in some places.